driver/opc/reader.cpp

```diff
@@ -569,21 +573,39 @@ std::unique_ptr<task::Task> Reader::configure(
     ctx->setState({
         .task = task.key,
         .variant = "success",
-        .details = {
-            {"running", false}
+        .details = json{
+            {"running", false},
+            {"message", "Task configured successfully"}
         }
     });
-    return std::make_unique<Reader>(ctx, task, cfg, breaker_config, std::move(source), writer_cfg);
+    return std::make_unique<Reader>(ctx, task, cfg, breaker_config, std::move(source),
+                                    writer_cfg);
 }
 
 void Reader::exec(task::Command &cmd) {
-    if (cmd.type == "start") pipe.start();
+    if (cmd.type == "start") {
+        pipe.start();
+        ctx->setState({
+            .task = task.key,
+            .variant = "success",
+            .details = json{
+                {"running", true},
+                {"message", "Task started successfully"}
+            }
+        });
+    }
     else if (cmd.type == "stop") return stop();
```

Comment:
you got rid of the Reader::start() but kept Reader::stop() which are nearly identical. you should get rid of both start and stop, or keep start()

Reply:
I guess the reason for this is because task::Task has a pure virtual stop() because the task manager uses this when it needs to reconfigure a task so I would not remove stop()